bin/delete_pr_comment

```diff
@@ -0,0 +1,48 @@
+#!/bin/bash -eu
+
+# This script is used to delete comments on a pull request, and must be run in a buldkite PR step.
+#
+# Example:
+#   delete_pr_comment --github-token <token> --containing "part of a comment"
+
+# Check dependencies and print their versions for diagnosis purposes
+jq --version
+curl --version
+
+# Parse command line arguments
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --containing) COMMENT_CONTAINING="$2"; shift ;;
+        --github-token) GITHUB_TOKEN="$2"; shift ;;
+        *) echo "Unknown option: $1"; exit 1 ;;
+    esac
+    shift
+done
+
+function require_option() {
+    if [[ -z "$2" ]]; then
+        echo "Error: $1 is required"
+        exit 1
+    fi
+}
+
+require_option '--containing' "$COMMENT_CONTAINING"
+require_option '--github-token' "$GITHUB_TOKEN"
+
+GITHUB_URL="${BUILDKITE_PULL_REQUEST_REPO%.git}"
+GITHUB_USER=$(basename "$(dirname "$GITHUB_URL")")
+GITHUB_REPO=$(basename "$GITHUB_URL")
+
+COMMENT_IDS="$(curl -s \
+    -H "Authorization: token ${GITHUB_TOKEN}" \
```

Comment:
Just curious. Is there any advantage in interpolating with `${GITHUB_TOKEN}` over `$GITHUB_TOKEN`?

Reply:
I guess `${}` is less prone to error, since it clearly states what the variable is. Like `${GITHUB_TOKEN}_VALUE` vs `$GITHUB_TOKEN_VALUE`. I think I copied and pasted a few code here and there, and simply forgot to make them consistent. 🥲